File: frontend/src/components/inventory/catalog/inventoryBehavior.js
export const INVENTORY_CLASS = {
  STOCK: "STOCK",
  EQUIPMENT: "EQUIPMENT",
};

export const DEFAULT_STOCK_ITEM_TYPE = "REAGENT";
export const EQUIPMENT_ITEM_TYPE = "EQUIPMENT";

export const INVENTORY_CLASS_OPTIONS = [
  {
    id: INVENTORY_CLASS.STOCK,
    text: "Stock item",
  },
  {
    id: INVENTORY_CLASS.EQUIPMENT,
    text: "Permanent equipment",
  },
];

export const getInventoryClass = (itemType) =>
  itemType === EQUIPMENT_ITEM_TYPE
    ? INVENTORY_CLASS.EQUIPMENT
    : INVENTORY_CLASS.STOCK;

export const isPermanentEquipment = (itemOrType) => {
  const itemType =
    typeof itemOrType === "string" ? itemOrType : itemOrType?.itemType;
  return getInventoryClass(itemType) === INVENTORY_CLASS.EQUIPMENT;
};

export const isStockManaged = (itemOrType) => !isPermanentEquipment(itemOrType);

export const isLotReceivable = (itemOrType) => isStockManaged(itemOrType);

File: src/main/java/org/openelisglobal/inventory/service/InventoryBehavior.java
package org.openelisglobal.inventory.service;

import org.openelisglobal.inventory.valueholder.InventoryEnums.ItemType;
import org.openelisglobal.inventory.valueholder.InventoryItem;

public final class InventoryBehavior {

    private InventoryBehavior() {
    }

    public static boolean isPermanentEquipment(InventoryItem item) {
        return item != null && isPermanentEquipment(item.getItemType());
    }

    public static boolean isPermanentEquipment(ItemType itemType) {
        return itemType == ItemType.EQUIPMENT;
    }

    public static boolean isStockManaged(InventoryItem item) {
        return item != null && isStockManaged(item.getItemType());
    }

    public static boolean isStockManaged(ItemType itemType) {
        return itemType != null && !isPermanentEquipment(itemType);
    }

    public static boolean isLotReceivable(InventoryItem item) {
        return isStockManaged(item);
    }
}
